File: sources/easygenerator.Web/App/mappers/singleSelectImageAnswerMapper.js
define(['models/singleSelectImageAnswer'],
    function (SingleSelectImageAnswerModel) {
        "use strict";

        var
            map = function (answer) {
                return new SingleSelectImageAnswerModel({
                    id: answer.Id,
                    image: answer.Image,
                    createdOn: new Date(answer.CreatedOn),
                    modifiedOn: new Date(answer.ModifiedOn),
                });
            };

        return {
            map: map
        };
    });

File: sources/easygenerator.Web/Components/Mappers/SingleSelectImageAnswerMapper.cs
﻿using easygenerator.DomainModel.Entities.Questions;
using easygenerator.Web.Extensions;

namespace easygenerator.Web.Components.Mappers
{
    public class SingleSelectImageAnswerMapper : EntityModelMapper<SingleSelectImageAnswer>
    {
        public override dynamic Map(SingleSelectImageAnswer entity)
        {
            return new
            {
                Id = entity.Id.ToNString(),
                Image = entity.Image,
                CreatedOn = entity.CreatedOn,
                CreatedBy = entity.CreatedBy,
                ModifiedOn = entity.ModifiedOn
            };
        }
    }
}